### PR TITLE
chore: initialize storybook using nx command

### PR DESCRIPTION
**Pull request by MFarabi619**
`nx add @nx/storybook`

## Summary by CodeRabbit

- **New Features**
	- Introduced new `Calendar` and `Accordion` components for enhanced user interaction.
	- Added a customizable `Button` component with various styles and sizes.
	- Implemented a `Typography` component suite for consistent text presentation.
	- New configuration files for Tailwind CSS and PostCSS to streamline styling processes.

- **Bug Fixes**
	- Updated `.gitignore` to improve clarity on ignored files.

- **Documentation**
	- Added README sections for shared UI library components and semantic color palette.

- **Chores**
	- Restructured project configurations and dependencies for better organization and efficiency.

### CODE REVIEW

**Comment by HasithDeAlwis** [comment]
Regarding Babel, all Storybook needs to work is to see a `vite.config.ts` file and internally it packages and builds itself from that, Vite handles it all for you, so a good starting point is just the boilerplate here and we optimize the configs as we go. 
```
import { defineConfig } from 'vite'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [],
})
```
I'm surprised you got it to work with Babel in the first place, that genuinely deserves an award :sob:

**Comment by MFarabi619** [comment]
> Regarding Babel, all Storybook needs to work is to see a `vite.config.ts` file and internally it packages and builds itself from that, Vite handles it all for you, so a good starting point is just the boilerplate here and we optimize the configs as we go.
> 
> ```
> import { defineConfig } from 'vite'
> 
> // https://vitejs.dev/config/
> export default defineConfig({
>   plugins: [],
> })
> ```
> 
> I'm surprised you got it to work with Babel in the first place, that genuinely deserves an award 😭

Handed off to you and Jeremy as per discord messages, feel free to take it out. I don't see us needing Babel anywhere else, and this was scaffolded by the nx storybook command I used lmao.

**Comment by MFarabi619** [comment]
Closing as stale

File: apps/website/app/routes/_index.tsx [hunk 1]
```diff
@@ -1,17 +1,42 @@
 import type { MetaFunction } from '@netlify/remix-runtime'
-import NxWelcome from '../nx-welcome'
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from '@shadcn/ui/accordion'
+import { Calendar } from '@shadcn/ui/calendar'
```
[the remark this answers is not in the document]
Comment (MFarabi619) [reply apps/website/app/routes/_index.tsx:8]: This is correct. `@nx-extend/shadcn-ui` is an nx plugin that fetches shadcn components and puts them into `libs/shared/ui`. `@shadcn/ui` is a local library. You can find the path alias for this in `tsconfig.base.json` at the root of the repo.

[the remark this answers is not in the document]
Comment (MFarabi619) [reply apps/website/app/routes/_index.tsx:8]: No I was replying to you

**Pull request merged**